fil_actor_interface/src/builtin/reward/mod.rs

```diff
@@ -100,4 +114,115 @@ impl State {
             State::V13(st) => from_token_v4_to_v2(st.into_total_storage_power_reward()),
         }
     }
+
+    /// The baseline power the network is targeting at this state's epoch.
+    pub fn this_epoch_baseline_power(self) -> StoragePower {
+        match self {
+            State::V8(st) => st.this_epoch_baseline_power,
+            State::V9(st) => st.this_epoch_baseline_power,
+            State::V10(st) => st.this_epoch_baseline_power,
+            State::V11(st) => st.this_epoch_baseline_power,
+            State::V12(st) => st.this_epoch_baseline_power,
+            State::V13(st) => st.this_epoch_baseline_power,
+        }
+    }
+
+    fn deal_provider_collateral_bounds_pre_v11(
```

Comment:
Why is the code here so different than the one `v12`+? Is the logic different?

Reply:
The logic is the same, it just does not exist in the original repo, so we have to introduce a shared helper. I'll add a comment.

Reply:
If the logic is the same, let's use the same, battle-tested code.

Reply:
I'd prefer to have a more readable version to whatever we copy-pasted from the original repo. However, it's done.

Reply:
You're welcome to propose a change in the upstream https://github.com/filecoin-project/builtin-actors to address readability.